### PR TITLE
Add JSDocOverloadTag to ForEachChildTable

### PR DESCRIPTION
**Pull request by a-tarasyuk**
@sandersn Do we need to add `JSDocOverloadTag` to `ForEachChildTable`?

### CODE REVIEW

**Comment by jakebailey** [comment]
Yes, I'm pretty sure you do. @sandersn

**Comment by jakebailey** [comment]
Hm, very interested in what baselines changed.

**Comment by a-tarasyuk** [comment]
@jakebailey Yes. I wonder why this tag was omitted in `ForEachChildTable`.

**Comment by a-tarasyuk** [comment]
@jakebailey fixed tests :)

File: src/compiler/utilitiesPublic.ts [hunk 1]
```diff
@@ -76,10 +76,11 @@ import {
     getElementOrPropertyAccessArgumentExpressionOrName,
     getEmitScriptTarget,
     getJSDocCommentsAndTags,
+    getJSDocRoot,
     getJSDocTypeParameterDeclarations,
     hasAccessorModifier,
-    hasDecorators,
     HasDecorators,
+    hasDecorators,
```
Comment (jakebailey) [inline src/compiler/utilitiesPublic.ts:83]: What re-sorted this? Hoping to pin down these so we don't keep flipping back and forth.

Comment (a-tarasyuk) [reply src/compiler/utilitiesPublic.ts:83]: I think `lint --fix` does that

Comment (jakebailey) [reply src/compiler/utilitiesPublic.ts:83]: Confusing, given it doesn't complain when it's sorted a specific way...

Comment (a-tarasyuk) [reply src/compiler/utilitiesPublic.ts]: I think to reproduce this issue need to do the following steps
1. Add some items to the file (`length`, `getJSDocRoot` etc.)
2. Run **QF** to add non-imported items
3. Run `lint --fix`

**Review by jakebailey — APPROVED** [review]

File: src/compiler/utilitiesPublic.ts [hunk 2]
```diff
@@ -1211,12 +1213,10 @@ function formatJSDocLink(link: JSDocLink | JSDocLinkCode | JSDocLinkPlain) {
  */
 export function getEffectiveTypeParameterDeclarations(node: DeclarationWithTypeParameters): readonly TypeParameterDeclaration[] {
     if (isJSDocSignature(node)) {
-        if (isJSDoc(node.parent)) {
-            const overloadTag = find(node.parent.tags, (tag) => {
-                return isJSDocOverloadTag(tag) && tag.typeExpression === node;
-            });
-            if (overloadTag) {
-                return flatMap(node.parent.tags, tag => isJSDocTemplateTag(tag) ? tag.typeParameters : undefined);
+        if (isJSDocOverloadTag(node.parent)) {
```
Comment (sandersn) [inline src/compiler/utilitiesPublic.ts:1216]: Nice fix. Reviewing the old code, I thought JSDocSignature was a supertype of JSDocOverload. But it's actually a child property, so `node.parent` of a JSDocSignature isn't the containing JSDoc, but a JSDocOverloadTag.

**Review by sandersn — APPROVED** [review]

**Pull request merged**